### PR TITLE
Update copyright header for 2017

### LINKED ISSUE #512: Thankfully, 2016 is over
Spotted while reviewing #511. Observe the copyright statement in the new file:

```python
# Most of this work is copyright (C) 2013-2016 David R. MacIver
# (david@drmaciver.com), but it contains contributions by others. See
```

We should change it to say 2017.

### PR DESCRIPTION
**Pull request by alexwlchan**
Just the result of running the `enforce_header.py` script over the entire codebase. Includes one fix for that script to avoid causing linting errors.

Resolves #512.

### CODE REVIEW

File: scripts/enforce_header.py [hunk 1]
```diff
@@ -37,8 +37,9 @@ def main():
                 o.write(shebang)
                 o.write("\n")
             o.write(HEADER_SOURCE)
-            o.write("\n\n")
-            o.write(source)
+            if source:
```
Comment (DRMacIver) [inline scripts/enforce_header.py]: What was this change about?

Comment (DRMacIver) [inline scripts/enforce_header.py]: Never mind, it makes sense if I actually read the commit it's from.

**Comment by DRMacIver** [comment]
LGTM, but you'll need to rerun against master.

**Comment by alexwlchan** [comment]
Rebased against master, new copyright year is now picked up dynamically.

File: scripts/enforce_header.py [hunk 2]
```diff
@@ -1,9 +1,12 @@
+from datetime import datetime
 import os
 import sys
 
 HEADER_FILE = "scripts/header.py"
 
-HEADER_SOURCE = open(HEADER_FILE).read().strip()
+CURRENT_YEAR = datetime.now().year
```
Comment (DRMacIver) [inline scripts/enforce_header.py]: This should probably be datetime.utcnow()

**Comment by alexwlchan** [comment]
> This should probably be datetime.utcnow()

Resolved.

**Pull request merged**